src/sdl3/pixels.rs

```diff
@@ -226,15 +266,104 @@ pub struct PixelMasks {
+    /// Bytes per pixel (padded depth)
+    pub bytes_per_pixel: u8,
+    /// Red mask
+    pub rmask: u32,
+    /// Green mask
+    pub gmask: u32,
+    /// Blue mask
+    pub bmask: u32,
+    /// Alpha mask
+    pub amask: u32,
+    /// Number of red bits
+    pub rbits: u8,
+    /// Number of green bits
+    pub gbits: u8,
+    /// Number of blue bits
+    pub bbits: u8,
+    /// Number of alpha bits
+    pub abits: u8,
+    /// Red shift count
+    pub rshift: u8,
+    /// Green shift count
+    pub gshift: u8,
+    /// Blue shift count
+    pub bshift: u8,
+    /// Alpha shift count
+    pub ashift: u8,
+}
 
 /// A pixel format, i.e. a set of masks that define how to pack and unpack pixel data.
 /// This is used to convert between pixel data and surface data.
 /// It wraps an SDL_PixelFormat.
-#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum PixelFormatEnum {
+    Unknown = SDL_PIXELFORMAT_UNKNOWN.0 as isize,
+    Index1LSB = SDL_PIXELFORMAT_INDEX1LSB.0 as isize,
+    Index1MSB = SDL_PIXELFORMAT_INDEX1MSB.0 as isize,
+    Index4LSB = SDL_PIXELFORMAT_INDEX4LSB.0 as isize,
+    Index4MSB = SDL_PIXELFORMAT_INDEX4MSB.0 as isize,
+    Index8 = SDL_PIXELFORMAT_INDEX8.0 as isize,
+    RGB332 = SDL_PIXELFORMAT_RGB332.0 as isize,
+    ARGB4444 = SDL_PIXELFORMAT_ARGB4444.0 as isize,
+    RGBA4444 = SDL_PIXELFORMAT_RGBA4444.0 as isize,
+    ABGR4444 = SDL_PIXELFORMAT_ABGR4444.0 as isize,
+    BGRA4444 = SDL_PIXELFORMAT_BGRA4444.0 as isize,
+    ARGB1555 = SDL_PIXELFORMAT_ARGB1555.0 as isize,
+    RGBA5551 = SDL_PIXELFORMAT_RGBA5551.0 as isize,
+    ABGR1555 = SDL_PIXELFORMAT_ABGR1555.0 as isize,
+    BGRA5551 = SDL_PIXELFORMAT_BGRA5551.0 as isize,
+    RGB565 = SDL_PIXELFORMAT_RGB565.0 as isize,
+    BGR565 = SDL_PIXELFORMAT_BGR565.0 as isize,
+    RGB24 = SDL_PIXELFORMAT_RGB24.0 as isize,
+    BGR24 = SDL_PIXELFORMAT_BGR24.0 as isize,
+    RGBX8888 = SDL_PIXELFORMAT_RGBX8888.0 as isize,
+    BGRX8888 = SDL_PIXELFORMAT_BGRX8888.0 as isize,
+    ARGB8888 = SDL_PIXELFORMAT_ARGB8888.0 as isize,
+    RGBA8888 = SDL_PIXELFORMAT_RGBA8888.0 as isize,
+    ABGR8888 = SDL_PIXELFORMAT_ABGR8888.0 as isize,
+    BGRA8888 = SDL_PIXELFORMAT_BGRA8888.0 as isize,
+    ARGB2101010 = SDL_PIXELFORMAT_ARGB2101010.0 as isize,
+    NV12 = SDL_PIXELFORMAT_NV12.0 as isize,
+    NV21 = SDL_PIXELFORMAT_NV21.0 as isize,
+    YV12 = SDL_PIXELFORMAT_YV12.0 as isize,
+    IYUV = SDL_PIXELFORMAT_IYUV.0 as isize,
+    YUY2 = SDL_PIXELFORMAT_YUY2.0 as isize,
+    UYVY = SDL_PIXELFORMAT_UYVY.0 as isize,
+    YVYU = SDL_PIXELFORMAT_YVYU.0 as isize,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
 pub struct PixelFormat {
     raw: SDL_PixelFormat,
 }
 
+impl PixelFormatEnum {
+    /// Converts this enum to the raw SDL_PixelFormat value.
+    #[doc(alias = "SDL_PixelFormat")]
+    pub fn to_ll(self) -> SDL_PixelFormat {
+        SDL_PixelFormat(self as i32)
+    }
+
+    /// Constructs a `PixelFormat` wrapper for this format.
```

Comment:
Maybe `impl From<PixelFormatEnum> for PixelFormat` instead would be more aligned with the rest of this lib

Reply:
Actually, you have already added this From below. I'm in favor of just removing this and moving this code there. These two procedures are doing the same thing in the end